distributed/scheduler.py

```diff
@@ -2126,6 +2126,19 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
                 ws = min(worker_pool.values(), key=operator.attrgetter("occupancy"))
+                if ws._occupancy == 0:
+                    # special case to use round-robin; linear search
+                    # for next worker with zero occupancy (or just
+                    # land back where we started).
+                    wp_vals = worker_pool.values()
+                    wp_i: WorkerState
+                    start: Py_ssize_t = self._n_tasks % n_workers
+                    i: Py_ssize_t
+                    for i in range(n_workers):
+                        wp_i = wp_vals[i + start % n_workers]
+                        if wp_i._occupancy == 0:
```

Comment:
What happens if we never find a worker with zero occupancy?

Reply:
after the loop ends `ws` will just stay as the original value determined from the `min(...)` call at L2128 (which is guaranteed to have zero occupancy inside the `if`)